ldms/src/ldmsd/ldmsd.c

```diff
@@ -1814,23 +1815,37 @@ int main(int argc, char *argv[])
 			break;
 		case 'A':
 			/* (multiple) auth options */
-			lval = strtok(optarg, "=");
+			
+			dup_auth = strdup(optarg);
+			if (!dup_auth) {
+				printf("Not enough memory parsing options!\n");
+				free(dup_auth);
+				exit(1);
+			}
+			lval = strtok(dup_auth, "=");
 			if (!lval) {
-				printf("ERROR: Expecting -A name=value\n");
+				printf("ERROR: Expecting -A name=value. Got %s\n",
+					dup_auth);
+				free(dup_auth);
 				exit(1);
 			}
 			rval = strtok(NULL, "");
 			if (!rval) {
-				printf("ERROR: Expecting -A name=value\n");
+				printf("ERROR: Expecting -A name=value. Got %s\n",
+					dup_auth);
```

Comment:
I don't think this is doing what you expect. Is there anyway that this can fail unless dup_auth is not '\0' terminated?

Reply:
My understanding of Narates check here is that we are parsing " -A arg" where arg is expected to be of the form $x=$y. lval becomes $x (which if = is missing becomes the entire arg). If arg is of the form "$x=" or just "$x", then the 2nd call to strtok returns NULL, according to the man page and we will have rval=NULL.

Reply:
The dup_auth should be optarg I think.

Reply:
@tom95858 it is optarg. is your browser caching something? dup_auth is used in no print statements now.